File: regexlib/python_re2_test_file/regexlib_7394.py
# 7394
# (((^\s*)*\S+\s+)|(\S+)){1,5} 
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"!"*64+"!_1SLQ_2"

import re2 as re
from time import perf_counter

regex = """(((^\s*)*\S+\s+)|(\S+)){1,5} """
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "!" * i * 1 + "!_1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_7394.js
/* 7394
 * (((^\s*)*\S+\s+)|(\S+)){1,5} 
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"!"*64+"!_1SLQ_2"
 */
var REGEX = /(((^\s*)*\S+\s+)|(\S+)){1,5} /
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '!'.repeat(i*1) + '!_1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}